src/backend/recycledRevoker/scanAttestations.ts

```diff
@@ -0,0 +1,74 @@
+import type { AccountId32 } from '@polkadot/types/interfaces';
+
+import { CType, Did, DidUri } from '@kiltprotocol/sdk-js';
+
+// import { logger } from '../utilities/logger';
+
+import { subScanEventGenerator } from './subScan';
+
+// export type EventParams = [
+//   { type_name: 'AttesterOf'; value: `0x${string}` },
+//   { type_name: 'ClaimHashOf'; value: `0x${string}` },
+//   { type_name: 'CTypeHashOf'; value: `0x${string}` },
+//   { type_name: 'DelegationNodeIdOf'; value: `0x${string}` | null },
+// ];
+
+export interface ParametersEntry {
+  type: string;
+  type_name: string;
+  value: AccountId32 | `0x${string}` | null;
+}
+
+export interface AttestationInfo {
+  owner: DidUri;
+  claimHash: `0x${string}`;
+  cTypeId: `kilt:ctype:0x${string}`;
+  block: number;
+  createdAt: Date;
+  extrinsicHash: `0x${string}`;
+  state: 'valid' | 'revoked' | 'removed' | undefined;
+}
+export async function* scanAttestations(fromBlock: number) {
+  const eventGenerator = subScanEventGenerator(
+    'attestation',
+    'AttestationCreated',
+    fromBlock,
+  );
+
+  for await (const event of eventGenerator) {
+    const {
+      block,
+      blockTimestampMs,
+      params: paramsObject,
+      extrinsicHash,
+    } = event;
+    const createdAt = new Date(blockTimestampMs);
+
+    const parametersArray: ParametersEntry[] = [...paramsObject];
+
+    const attesterOf = parametersArray.find((element) => {
+      return element.type_name === 'AttesterOf';
+    })?.value as AccountId32;
+    const cTypeHash = parametersArray.find((element) => {
+      return element.type_name === 'CtypeHashOf';
+    })?.value as `0x${string}`;
+
+    const claimHash = parametersArray.find((element) => {
+      return element.type_name === 'ClaimHashOf';
+    })?.value as `0x${string}`;
```

Comment:
As far as I understand the blockchain, the order of these parameters is always the same, so we can hard-code the indexes, as in the provided code was doing before. Or did you find a case of a different order of parameters?

Reply:
This way it improves readability and it will break when subscan or kilt change something.

Reply:
As compared to the original code below, this definitely does not improve readability, do you agree @lukeg90?
```ts
    const owner = Did.fromChain(params[0].value);
    const claimHash = params[1].value;
    const cTypeId = CType.hashToId(params[2].value);
    const delegationId = params[3].value;

Reply:
params[1] does not tell me anything

Reply:
I agree

Reply:
done